webapp/channels/src/components/date_time_picker_modal/date_time_picker_modal.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {Moment} from 'moment-timezone';
+import React, {useCallback, useEffect, useState} from 'react';
+import {useSelector} from 'react-redux';
+
+import {GenericModal} from '@mattermost/components';
+
+import {getCurrentTimezone} from 'mattermost-redux/selectors/entities/timezone';
+
+import DateTimeInput, {getRoundedTime} from 'components/custom_status/date_time_input';
+
+import Constants from 'utils/constants';
+import {isKeyPressed} from 'utils/keyboard';
+import {getCurrentMomentForTimezone} from 'utils/timezone';
+
+import './style.scss';
+
+type Props = {
+    onExited: () => void;
+    ariaLabel: string;
+    header: React.ReactNode;
+    subheading?: React.ReactNode;
+    onChange?: (dateTime: Moment) => void;
+    onCancel?: () => void;
+    onConfirm?: (dateTime: Moment) => void;
+    initialTime?: Moment;
+    confirmButtonText?: React.ReactNode;
+    cancelButtonText?: React.ReactNode;
+    bodyPrefix?: React.ReactNode;
+    bodySuffix?: React.ReactNode;
+    relativeDate?: boolean;
+};
+
+export default function DateTimePickerModal({onExited, ariaLabel, header, onConfirm, onCancel, initialTime, confirmButtonText, onChange, cancelButtonText, subheading, bodyPrefix, bodySuffix, relativeDate}: Props) {
+    const userTimezone = useSelector(getCurrentTimezone);
+    const currentTime = getCurrentMomentForTimezone(userTimezone);
+    const initialRoundedTime = getRoundedTime(currentTime);
+
+    const [dateTime, setDateTime] = useState(initialTime || initialRoundedTime);
+
+    const [isDatePickerOpen, setIsDatePickerOpen] = useState(false);
+
+    useEffect(() => {
+        function handleKeyDown(event: KeyboardEvent) {
+            if (isKeyPressed(event, Constants.KeyCodes.ESCAPE) && !isDatePickerOpen) {
+                onExited();
+            }
+        }
```

Comment:
I guess in this case you want to prevent the default too? 0/5.

Reply:
I don't think its needed. Any reason for it?

Reply:
My thought is simply that since you are handling it here, it shouldn't be handled anywhere else. But no strong feelings with it.

In any case, if we add it, I meant to add it inside the `if`.

Reply:
Done.